tests/Doctrine/Tests/ORM/Functional/Ticket/DDC7561Test.php

```diff
@@ -0,0 +1,282 @@
+<?php
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\Tests\OrmFunctionalTestCase;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Types\ConversionException;
+use Doctrine\DBAL\Types\Type;
+
+/**
+ * @group DDC-1707
+ */
+class DDC7561Test extends OrmFunctionalTestCase
+{
+    public function setUp()
+    {
+        parent::setUp();
+
+        Type::addType(
+            'ddc_7561_withdraw_order_confirmation_status',
+            DDC7561DDC7561ConfirmationStatusType::class
+        );
+
+        $this->_schemaTool->createSchema(
+            [
+                $this->_em->getClassMetadata(DDC7561AbstractOrder::class),
+                $this->_em->getClassMetadata(DDC7561CardOrder::class),
+                $this->_em->getClassMetadata(DDC7561WithdrawOrder::class),
+            ]
+        );
+    }
+
+    public function testExpectsDDC7561CardOrderFetching()
+    {
+        $expectedOrder = new DDC7561CardOrder(1, '1000', 'USD');
+
+        $this->_em->persist($expectedOrder);
+        $this->_em->flush();
+        $this->_em->clear();
+
+        $repository = $this->_em->getRepository(DDC7561AbstractOrder::class);
+
+        $actualOrder = $repository->find(1);
+
+        $this->assertEquals($expectedOrder, $actualOrder);
+    }
+}
+
+/**
+ * @Entity
+ * @Table(name="ddc_7561_orders")
+ * @InheritanceType("JOINED")
+ * @DiscriminatorColumn(name="type", type="string")
+ * @DiscriminatorMap({
+ *     "card" = "DDC7561CardOrder",
+ *     "withdraw" = "DDC7561WithdrawOrder"
+ * })
+ */
+abstract class DDC7561AbstractOrder {
+    /**
+     * @Id()
+     * @GeneratedValue()
```

Comment:
The ID should be stable for our testing purposes: let's please use `strategy="NONE"`